app/controllers/ui_job_wizard_controller.rb

```diff
@@ -8,6 +8,14 @@ def categories
     render :json => {:job_categories =>job_categories}
   end
 
+  def template
+    job_template = JobTemplate.authorized.find(params[:id])
+    render :json => {
```

Comment:
in case the `job_template` doesn't exist, let's return another JSON with status `:not_found`

Reply:
It kinda works even now, but it would be nice if it could say `job_template` instead of `ui_job_wizard`

```
curl -i -u admin:changeme http://localhost:3000/ui_job_wizard/template/91                                                                                  
HTTP/1.1 404 Not Found
Foreman_version: 2.6.0-develop
Foreman_api_version: 2
Foreman_current_organization: ; ANY
Foreman_current_location: ; ANY
Content-Type: application/json; charset=utf-8
-----B<-----MORE HTTP HEADERS-----B<-----

{
  "error": {"message":"Resource ui_job_wizard not found by id '91'"}
}
```

Reply:
nice, though if I change it and access from the UI, it kinda crashes the page - at least on my end :)

Reply:
In the ui is shows as: 
![Screenshot from 2021-05-25 12-21-56](https://user-images.githubusercontent.com/30431079/119482148-32b07c80-bd5c-11eb-8015-b681f1d98830.png)
@adamruzicka, how can I change the resource name in the error?

Reply:
It is taken from `#resource_name` method on the controller. IIRC by default it guesses the name from the controller's name.

Reply:
The user can't access the next step if there is no template selected. A user can create job categories only in the job template form so there can't be empty categories, and if there are then it's an error.